LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -6,57 +6,42 @@ namespace LearningMissionSimulation
 {
     public class PlaygroundMher
     {
-        public PlaygroundMher()
-        {
-            
-        }
-       
-        //Account account = ObjectGenerator.GenerateAccount();
+        Queue<Account> accountQueue = new Queue<Account>();
 
-        List<Account> accountList = new List<Account>();
+        Dictionary<Guid, Account> accountDictionary = new Dictionary<Guid, Account>();
 
-        Queue<Status> statusQueue = new Queue<Status>();
-
-        // Simulate account creation  process
         public void CreateAccounts(int accountCount)
         {
             int i = 0;
             while (i < accountCount)
             {
-                accountList.Add(ObjectGenerator.GenerateAccount());
-                i++;
-            }
-            i = 0;
-            foreach (var account in accountList)
-            {
-                if (account.Status == Status.Pending)
+                Account account = ObjectGenerator.GenerateAccount();
+                accountDictionary.Add(account.Id, account);
+                if (account.Role == Role.Student || account.Role == Role.Instructor)
```

Comment:
You have to add also a filter by status, only accounts with status pending should be added to your queue.